app/controllers/habits_controller.rb

```diff
@@ -0,0 +1,53 @@
+class HabitsController < ApplicationController
+  before_action :authorize_request
+  before_action :find_user
+
+  def index
+    @habits = @user.habits.all
+    render json: @habits, status: :ok  
+  end
+
+  def create
+    @habit = @user.habits.create!(habit_params)
```

Comment:
Calling `create!` here will actually create the record, which makes the `save` call below unnecessary. I would go with `@habit = @user.habits.new(habit_params)`

Also, you don't need to use instance variables in most of these cases since you're not using these variables in a template.

Reply:
When I switch to new, the record on the join table is not created, only when I use create